ios/sound.mm

```diff
@@ -24,209 +24,279 @@
 #include "sound.h"
 #include <AVFoundation/AVFoundation.h>
 
+#define kOutputBus 0
+#define kInputBus  1
 
 /* Implementation *************************************************************/
-CSound::CSound ( void           (*fpNewProcessCallback) ( CVector<short>& psData, void* arg ),
+CSound::CSound ( void ( *fpNewProcessCallback ) ( CVector<short>& psData, void* arg ),
                  void*          arg,
                  const QString& strMIDISetup,
-                 const bool     ,
+                 const bool,
                  const QString& ) :
     CSoundBase ( "CoreAudio iOS", fpNewProcessCallback, arg, strMIDISetup ),
-    midiInPortRef ( static_cast<MIDIPortRef> ( NULL ) )
+    isInitialized ( false )
 {
-    NSError *audioSessionError = nil;
-  
-    [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
-    [[AVAudioSession sharedInstance] requestRecordPermission:^(BOOL granted) {
-        if (granted) {
-            // ok
-        }
-    }];
-    [[AVAudioSession sharedInstance] setMode:AVAudioSessionModeMeasurement error:&audioSessionError];
+    try
+    {
+        NSError* audioSessionError = nil;
 
-  //GetAvailableInOutDevices();
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
+        [[AVAudioSession sharedInstance] requestRecordPermission:^( BOOL granted ) {
+            if ( granted )
+            {
+                // ok
+            }
+            else
+            {
+                //TODO - signal to CClient?
+            }
+        }];
+        [[AVAudioSession sharedInstance] setMode:AVAudioSessionModeMeasurement error:&audioSessionError];
+    }
+    catch ( const CGenErr& generr )
+    {
+        qDebug ( "Sound exception ...." );
```

Comment:
This is not very useful... at least we should output what the error was.

Reply:
Ok will do